mmv1/products/gkeonprem/BareMetalNodePool.yaml

```diff
@@ -0,0 +1,248 @@
+# limitations under the License.
+--- !ruby/object:Api::Resource
+name: "BareMetalNodePool"
+min_version: beta
+base_url: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{bare_metal_cluster}}/bareMetalNodePools"
+create_url: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{bare_metal_cluster}}/bareMetalNodePools?bare_metal_node_pool_id={{name}}"
+update_url: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{bare_metal_cluster}}/bareMetalNodePools/{{name}}"
+self_link: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{bare_metal_cluster}}/bareMetalNodePools/{{name}}"
+update_verb: :PATCH
+update_mask: true
+description: "A Google Bare Metal Node Pool."
+autogen_async: false
+id_format: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{bare_metal_cluster}}/bareMetalNodePools/{{name}}"
+import_format: ["projects/{{project}}/locations/{{location}}/bareMetalClusters/{{bare_metal_cluster}}/bareMetalNodePools/{{name}}"]
+taint_resource_on_failed_create: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "gkeonprem_bare_metal_node_pool_basic"
+    primary_resource_id: "nodepool-basic"
+    min_version: beta
+    vars:
+      name: "nodepool"
+    test_env_vars:
+      project: "fake-backend-360322"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "gkeonprem_bare_metal_node_pool_full"
+    primary_resource_id: "nodepool-full"
+    min_version: beta
+    vars:
+      name: "nodepool"
+    test_env_vars:
+      project: "fake-backend-360322"
+skip_sweeper: true
+parameters:
+  - !ruby/object:Api::Type::String
+    name: "name"
+    description: The bare metal node pool name.
+    immutable: true
+    url_param_only: true
+    required: true
+  - !ruby/object:Api::Type::ResourceRef
+    name: "bareMetalCluster"
+    resource: "BareMetalCluster"
+    imports: "name"
+    immutable: true
+    description: "The cluster this node pool belongs to."
+    url_param_only: true
+    required: true
+  - !ruby/object:Api::Type::String
+    name: "location"
+    description: The location of the resource.
+    immutable: true
+    url_param_only: true
+    required: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: "displayName"
+    description: |
+      The display name for the Bare Metal Node Pool.
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: "annotations"
+    description: |
+      Annotations on the Bare Metal Node Pool.
+      This field has the same restrictions as Kubernetes annotations.
+      The total size of all keys and values combined is limited to 256k.
+      Key can have 2 segments: prefix (optional) and name (required),
+      separated by a slash (/).
+      Prefix must be a DNS subdomain.
+      Name must be 63 characters or less, begin and end with alphanumerics,
+      with dashes (-), underscores (_), dots (.), and alphanumerics between.
+    default_from_api: true
```

Comment:
Same question here for default_from_api

Reply:
Yes, API will add some default annotations automatically on server side. We need to mark it as `default_from_api` to avoid unintentional override on this field during update.